### ISSUE TITLE
[Accessibility]: Branch/tag/commit + Repositories selector: Incorrect keyboard navigation

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35474]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34417, https://github.com/sourcegraph/sourcegraph/issues/34416

### Problem description

<img width="608" alt="image" src="https://user-images.githubusercontent.com/9516420/168596380-b218979a-a672-4abe-ae09-8a7e117e1c0a.png">

We should be able to use arrow keys to navigate up/down the list - we can only use tab keys. This makes it difficult to navigate out of the list, if you are half way in.

### Expected behavior

It should function like a menu: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role

We should be able to give this `menu` and `menuitem` roles to enable proper keyboard controls.

### Additional details

The search context selector is an example where this behavior works well

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
Also present on repo selector, updated issue
